packages/mdc-slider/adapter.ts

```diff
@@ -21,178 +21,138 @@
  * THE SOFTWARE.
  */
 
-/* eslint-disable no-unused-vars */
-
 /**
- * Adapter for MDC Slider.
- *
- * Defines the shape of the adapter expected by the foundation. Implement this
- * adapter to integrate the Slider into your framework. See
- * https://github.com/material-components/material-components-web/blob/master/docs/authoring-components.md
- * for more information.
- *
- * @record
+ * Defines the shape of the adapter expected by the foundation.
+ * Implement this adapter for your framework of choice to delegate updates to
+ * the component in your framework of choice. See architecture documentation
+ * for more details.
+ * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
  */
-class MDCSliderAdapter {
+interface MDCSliderAdapter {
   /**
    * Returns true if className exists for the slider Element
-   * @param {string} className
-   * @return {boolean}
    */
-  hasClass(className) {}
+  hasClass(className: string): boolean;
 
   /**
    * Adds a class to the slider Element
-   * @param {string} className
    */
-  addClass(className) {}
+  addClass(className: string): void;
 
   /**
    * Removes a class from the slider Element
-   * @param {string} className
    */
-  removeClass(className) {}
+  removeClass(className: string): void;
 
   /**
-   * Returns a string if attribute name exists on the slider Element,
-   * otherwise returns null
-   * @param {string} name
-   * @return {?string}
+   * Returns a string if attribute name exists on the slider Element, otherwise returns null
    */
-  getAttribute(name) {}
+  getAttribute(name: string): string | null;
 
   /**
    * Sets attribute name on slider Element to value
-   * @param {string} name
-   * @param {string} value
    */
-  setAttribute(name, value) {}
+  setAttribute(name: string, value: string): void;
 
   /**
    * Removes attribute name from slider Element
-   * @param {string} name
    */
-  removeAttribute(name) {}
+  removeAttribute(name: string): void;
 
   /**
    * Returns the bounding client rect for the slider Element
-   * @return {?ClientRect}
    */
-  computeBoundingRect() {}
+  computeBoundingRect(): ClientRect | null;
 
   /**
    * Returns the tab index of the slider Element
-   * @return {number}
    */
-  getTabIndex() {}
+  getTabIndex(): number;
 
   /**
    * Registers an event handler on the root element for a given event.
-   * @param {string} type
-   * @param {function(!Event): undefined} handler
    */
-  registerInteractionHandler(type, handler) {}
+  registerInteractionHandler(type: string, handler: EventListener): void;
```

Comment:
all of the `string` and `EventListener` should be `EventType` and `SpecificEventListener`

Reply:
Done